shared/api/certificate.go

```diff
@@ -72,3 +77,44 @@ type Certificate struct {
 func (cert *Certificate) Writable() CertificatePut {
 	return cert.CertificatePut
 }
+
+// CertificateAddToken represents the fields contained within an encoded client join token.
+//
+// swagger:model
+//
+// API extension: certificate_token
+type CertificateAddToken struct {
+	// The name of the new client
+	// Example: user@host
+	ClientName string `json:"client_name" yaml:"client_name"`
+
+	// The fingerprint of the network certificate
+	// Example: 57bb0ff4340b5bb28517e062023101adf788c37846dc8b619eb2c3cb4ef29436
+	Fingerprint string `json:"fingerprint" yaml:"fingerprint"`
+
+	// The addresses of the server
+	// Example: ["10.98.30.229:8443"]
+	Addresses []string `json:"addresses" yaml:"addresses"`
+
+	// The random join secret.
+	// Example: 2b2284d44db32675923fe0d2020477e0e9be11801ff70c435e032b97028c35cd
+	Secret string `json:"secret" yaml:"secret"`
+
+	// List of allowed projects (applies when restricted)
+	// Example: ["default", "foo", "bar"]
+	Projects []string `json:"projects" yaml:"projects"`
+
+	// Whether to limit the certificate to listed projects
+	// Example: true
+	Restricted bool `json:"restricted" yaml:"restricted"`
```

Comment:
This should not be in the add token.

The add token should only include what the client needs. The list of projects and whether it's restricted is something the administrator has set and that should remain on the server side.